### ISSUE TITLE
Failed to create VM in the enviroment with the devicemapper as the storage driver of Docker

### ISSUE THREAD

**Issue by luckyfengyong** [kubevirt/kubevirt#3117]
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
/kind bug
> /kind enhancement

**What happened**:
In the enviroment with the devicemapper as the storage driver of Docker, VMI failed to be started with error message of `failed to detect root mount info of containerDisk` in the log of `virt-hanlder`. 

The root cause probably is that `virt-handler` assume in the pods of `virt-launcher`, the rootfs of `/` is mount from `/` which is true with the overlay fs as the storage driver of Docker. However, when devicemapper is configured as the storage driver of Docker, it is mouted from `/rootfs`. 

In /proce/$pid/mountinfo, it is shown as something like

```
...
1626 850 253:29 /rootfs / rw,relatime - ext4 /dev/mapper/docker-253:2- .....
...
```
With a fix to allow `/rootfs` to be recognized as root mount info of containerDisk, the issue is resolved and VM is created successfully.

**What you expected to happen**:

Create VM successfully in the enviroment with the devicemapper as the storage driver of Docker

**How to reproduce it (as minimally and precisely as possible)**:
Configure devicemapper as the storage driver of Docker and then create VMI according to kube-virt user doc.

**Anything else we need to know?**:

**Environment**:
- KubeVirt version (use `virtctl version`): v0.26.0
- Kubernetes version (use `kubectl version`): v1.15.0
- VM or VMI specifications: 
```
apiVersion: kubevirt.io/v1alpha3
kind: VirtualMachineInstance
metadata:
  name: testvmi-yong1
  namespace: fst-manage
spec:
  terminationGracePeriodSeconds: 30
  domain:
    clock:
      timezone: "Asia/Shanghai"
      timer: {}
    resources:
      requests:
        cpu: 2
        memory: 2048M
    devices:
      disks:
      - name: containerdisk
        disk: 
          bus: virtio
      - name: emptydisk
        disk:
          bus: virtio
      - name: cloudinitdisk
        disk:
          bus: virtio
  volumes:
  - name: containerdisk
    containerDisk:
      image: 192.39.12.230:20202/kubevirt/fedora-cloud-container-disk-demo:latest
  - name: emptydisk
    emptyDisk:
      capacity: "2Gi"
  - name: cloudinitdisk
    cloudInitNoCloud:
      userData: |-
        #cloud-config
        password: fedora
        chpasswd: { expire: False }
```
- Cloud provider or hardware configuration: N/A
- OS (e.g. from /etc/os-release): centos
- Kernel (e.g. `uname -a`): 3.10.0

**Comment by rmohr** [comment]
Nice catch. If you want this fixed fast, it would be awesome if you could create a PR.

**Comment by dren-dk** [comment]
It seems this problem is closely related to #3119 which happens on btrfs.

This is a shame as I was considering switching to LVM as a workaround for my btrfs woes.

The reason I want either LVM or btrfs is that I have a workload that contains huge data files that get sparsely written to during execution, so block-level COW in the overlay file system is very much needed.

**Comment by rmohr** [comment]
I think #3124 should solve the issue for all filesystem backends. I think that my validation was just too strict.

**Comment by luckyfengyong** [comment]
@rmohr #3124 seems not fix all the issue. Besides checking in `MountInfoRoot()`, we also need fix following codes in  `Mount()`  to add `nodeMountInfo.Root` into `sourceFile` as well.

```
				sourceFile, err := containerdisk.GetImage(filepath.Join(nodeRes.MountRoot(), nodeMountInfo.MountPoint), volume.ContainerDisk.Path)
				if err != nil {
					return fmt.Errorf("failed to find a sourceFile in containerDisk %v: %v", volume.Name, err)
				}
```

**Comment by rmohr** [comment]
@luckyfengyong could you share `/proc/<pid>/mountinfo` for the virt-launcher pid and pid `1`? What is then the final full path of the mount?

**Comment by luckyfengyong** [comment]
@rmohr Sorry, the enviroment was reprovisioned for other purpose. But I keep the related error message as follows.

`/proc/1/root/opt/docker/devicemapper/mnt/83f8464c5ce21fa7334ed8d181d235c935c41421f347c2d03d3347abe2488171/disk: no such file or directory","timestamp":"2020-02-29T07:12:59.562971Z","uid":"cba8f66a-b1dc-46ef-bc0f-2fb1e45e828f"`

The right path is `/proc/1/root/opt/docker/devicemapper/mnt/83f8464c5ce21fa7334ed8d181d235c935c41421f347c2d03d3347abe2488171/rootfs/disk`

BTW, when I opened the issue, I appended the item of `/rootfs` from `/proc/<pid>/mountinfo` of the virt-launcher pid as follows. 
```
...
1626 850 253:29 /rootfs / rw,relatime - ext4 /dev/mapper/docker-253:2- .....
...
```
I will recreate the env and append the requested info later.
